Champions/Caitlyn/E.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using LeagueSandbox.GameServer.Logic.API;
+using LeagueSandbox.GameServer.Logic.GameObjects;
+using LeagueSandbox.GameServer.Logic.Scripting.CSharp;
+using System.Numerics;
+
+namespace Spells
+{
+    public class CaitlynEntrapment : GameScript
+    {
+        public void OnActivate(Champion owner) { }
+
+        public void OnDeactivate(Champion owner) { }
+
+        public void OnStartCasting(Champion owner, Spell spell, Unit target) {
+            spell.spellAnimation("Spell3", owner);
+        }
+
+        public void OnFinishCasting(Champion owner, Spell spell, Unit target) {
+            // Calculate net coords
+            var current = new Vector2(owner.X, owner.Y);
+            var to = Vector2.Normalize(new Vector2(spell.X, spell.Y) - current);
+            var range = to * 750;
+            var trueCoords = current + range;
+
+            // Calculate dash coords/vector
+            var dash = Vector2.Negate(to) * 500;
+            var dashCoords = current + dash;
+            ApiFunctionManager.DashToLocation(owner, dashCoords.X, dashCoords.Y, 1000, true, "Spell3b");
+            spell.AddProjectile("CaitlynEntrapmentMissile", trueCoords.X, trueCoords.Y);
+            spell.spellAnimation("Spell3_fallback", owner);
+        }
+
+        public void ApplyEffects(Champion owner, Unit target, Spell spell, Projectile projectile) {
+            var ap = owner.GetStats().AbilityPower.Total * 0.8f;
+            var damage = 80 + spell.Level * 50 + ap;
+            owner.DealDamageTo(target, damage, DamageType.DAMAGE_TYPE_MAGICAL, DamageSource.DAMAGE_SOURCE_SPELL, false);
+
+            var slowDuration = 1.0f;
+            switch (spell.Level)
+            {
+                case 1:
+                    slowDuration = 1;
+                    break;
+                case 2:
+                    slowDuration = 1.25f;
+                    break;
+                case 3:
+                    slowDuration = 1.5f;
+                    break;
+                case 4:
+                    slowDuration = 1.75f;
+                    break;
+                case 5:
+                    slowDuration = 2;
+                    break;
+                default:
+                    break;
+            }
```

Comment:
Might be cleaner to have this as either an array index `new int[] { 0, 1, 1.25, 1.5, 1.75, 2}[spell.Level]`, or as a simple mathematical calculation (`1 + (spell.Level - 1) * 0.25`).

Reply:
What molenz suggested + a dictionary works too